programs/mpl-core/src/plugins/utils.rs

```diff
@@ -621,13 +621,14 @@ pub fn delete_plugin<'a, T: DataBlob>(
             .checked_sub(next_plugin_offset)
             .ok_or(MplCoreError::NumericalOverflow)?;
 
-        //TODO: This is memory intensive, we should use memmove instead probably.
-        let src = account.data.borrow()[next_plugin_offset..].to_vec();
-        sol_memcpy(
-            &mut account.data.borrow_mut()[plugin_offset..],
-            &src,
-            data_to_move,
-        );
+        unsafe {
+            let base = account.data.borrow_mut().as_mut_ptr();
+            sol_memmove(
+                base.add(plugin_offset),
+                base.add(next_plugin_offset),
+                data_to_move,
+            );
+        }
```

Comment:
This is good and I think we should do it, but should we add some checks?

One convention I've seen before is a safety comment required above unsafe blocks, i.e:
```
// SAFETY: `borrow_mut` will always return a valid pointer,
// and we check that the `plugin_offset` and `next_plugin_offset`
// are within range, and that the size of `data_to_move` is will not
// make it overflow the bounds when it is moved to its new location.
unsafe {
   ...
}
```

Or
```
// SAFETY: `borrow_mut` will always return a valid pointer, and this will
// fail and revert state if there is a memory violation.
unsafe {
   ...
}
```

Reply:
Yes! Excellent idea.